Source/drlg_l3.cpp

```diff
@@ -1867,34 +1866,20 @@ bool PlaceAnvil()
 			break;
 	}
 
-	setpc_x = sx;
-	setpc_y = sy;
-	setpc_w = width + 2;
-	setpc_h = height + 2;
+	SetPiece = { { sx, sy }, { width + 2, height + 2 } };
 
-	for (int yy = 0; yy < setpc_h; yy++) {
-		for (int xx = 0; xx < setpc_w; xx++) {
+	for (int yy = 0; yy < SetPiece.size.width; yy++) {
+		for (int xx = 0; xx < SetPiece.size.height; xx++) {
 			Protected[xx + sx][yy + sy] = true;
 		}
 	}
 
-	sx++;
-	sy++;
-
-	const uint16_t *tileLayer = &dunData[2];
-	for (int yy = 0; yy < height; yy++) {
-		for (int xx = 0; xx < width; xx++) {
-			int tileId = tileLayer[xx + yy * width];
-			if (tileId != 0) {
-				dungeon[xx + sx][yy + sy] = tileId;
-			}
-		}
-	}
+	PlaceDunTiles(dunData.get(), { sx + 1, sy + 1 }, 7);

```

Comment:
Is this meant to be floorId = 0? The original loop didn't change dungeon if tileId was 0

Reply:
(3 of them do require that the floor tile is specified so it's not possible to always leave it out either)